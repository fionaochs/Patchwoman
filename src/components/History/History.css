.tealButton {
  background-color: #140567;
  font-family: 'Volkkorn', sans-serif;
  font-weight: bold;
  font-size: 1rem;
  color: white;
  border-radius: .3rem;
  border: none;
  height: 2.5rem;
  width: 10rem;
  margin-top: 2rem;
  letter-spacing: 0.02rem;
}
.HistoryBox {
  display: flex;
  flex-direction: row;
}
.History {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 45%;
  height: 50vh;
  
  button, li {
    cursor: pointer;
  }

  button:hover {
    background-color: #4238c4;
    cursor: pointer;
  }

  h3 {
    font-weight: 300;
  }

  li {
    list-style-type: none;
    margin: 1rem 0;
    vertical-align: middle;
    padding: 1rem 1.5rem;
    border-radius: .3rem;
    background-color: #d6eaf7;
    min-height: 6em;
    min-width: 20em;
    width: 80%;
  }
}