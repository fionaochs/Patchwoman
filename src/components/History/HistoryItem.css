.HistoryItem {
  h3 {
    text-transform: uppercase;
    color: white;
    margin: 0 auto;
  }

  p {
    font-weight: 400;
    font-size: 0.8rem;
  }
}