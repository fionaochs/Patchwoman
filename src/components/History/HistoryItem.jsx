import React from 'react';
import PropTypes from 'prop-types';
import styles from './HistoryItem.css';

const HistoryItem = ({ method, url, onClick }) => {
  return (
    <section className={styles.HistoryItem} onClick={onClick}>
      <h3>{method}</h3>
      <p>{url}</p>
    </section>
  );
};

HistoryItem.propTypes = {
  method: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired
};

export default HistoryItem;
