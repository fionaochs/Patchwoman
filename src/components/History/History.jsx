import React from 'react';
import PropTypes from 'prop-types';
import HistoryItem from './HistoryItem.jsx';
import { weakKey } from '../../utilities/weakKey.jsx';
import styles from './History.css';

const History = ({ requests, onClick, onDelete }) => {
  const requestElements = requests.map(request => (
    <li key={weakKey(request)}>
      <HistoryItem {...request} onClick={() => onClick(request)} />
    </li>
  ));
  
  return (
    <div className={styles.HistoryBox}>
      <ul className={styles.History}>
        {requestElements}
        <button className={styles.button} onClick={onDelete}>Clear History</button>
      </ul>
    </div>
  );
};

History.propTypes = {
  onClick: PropTypes.func.isRequired,
  onDelete: PropTypes.func.isRequired,
  requests: PropTypes.arrayOf(PropTypes.shape({
    method: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    body: PropTypes.string,
  }))
};

export default History;

