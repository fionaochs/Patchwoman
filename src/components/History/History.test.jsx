import React from 'react';
import { shallow } from 'enzyme';
import History from './History.jsx';

describe('RequestHistory', () => {
  it('matches a snapshot', () => {
    const requests = [{
      method: 'post', 
      url: 'https://jsonplaceholder.typicode.com/posts'
    }, {
      method: 'get', 
      url: 'https://jsonplaceholder.typicode.com/posts'
    }, {
      method: 'put', 
      url: 'https://jsonplaceholder.typicode.com/posts'
    }, {
      method: 'put', 
      url: 'https://jsonplaceholder.typicode.com/posts'
    }];

    const wrapper = shallow(
      <History requests={requests} onClick={() => {}}onDelete={() => {}} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
