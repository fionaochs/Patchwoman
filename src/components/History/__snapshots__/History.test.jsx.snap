// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestHistory matches a snapshot 1`] = `
<div
  className="HistoryBox"
>
  <ul
    className="History"
  >
    <li
      key="weak-key-0"
    >
      <HistoryItem
        method="post"
        onClick={[Function]}
        url="https://jsonplaceholder.typicode.com/posts"
      />
    </li>
    <li
      key="weak-key-1"
    >
      <HistoryItem
        method="get"
        onClick={[Function]}
        url="https://jsonplaceholder.typicode.com/posts"
      />
    </li>
    <li
      key="weak-key-2"
    >
      <HistoryItem
        method="put"
        onClick={[Function]}
        url="https://jsonplaceholder.typicode.com/posts"
      />
    </li>
    <li
      key="weak-key-3"
    >
      <HistoryItem
        method="put"
        onClick={[Function]}
        url="https://jsonplaceholder.typicode.com/posts"
      />
    </li>
    <button
      className="button"
      onClick={[Function]}
    >
      Clear History
    </button>
  </ul>
</div>
`;
